nikola/plugin_categories.py

```diff
@@ -75,8 +75,30 @@ def set_site(self, site):
     def inject_templates(self):
         """Inject 'templates/<engine>' (if exists) very early in the theme chain."""
         try:
+            modname = ""
+            possmodnames = (
+                self.__class__.__module__,
+                # since https://github.com/tibonihoo/yapsy/pull/11 ,
+                # yapsy only adds each imported plugin to sys.modules
+                # under its modified, "unique" name (see early in
+                # PluginManager.loadPlugins), so we recreate the
+                # modified name here to find it. we fudge the serial
+                # number here, assuming that if a plugin is loaded
+                # under the same name multiple times, the location
+                # will also be the same, so we can just use 0.
+                "yapsy_loaded_plugin_" + self.__class__.__module__ + "_0",
+                "yapsy_loaded_plugin_" + self.name + "_0",
+            )
+            for possmodname in possmodnames:
+                if possmodname in sys.modules:
+                    modname = possmodname
```

Comment:
Consider one less dictionary lookup:
```py
module_candidate = None
for possible_name in possible_names:
    module_candidate = sys.modules.get(possible_name)
    if module_candidate is not None:
        break
if module_candidate is None:
    return
```

Reply:
Yeah, I agree, I did the change to a loop quickly and it could be more elegant. I'll tweak it.